Added DataCite support

vjrj opened the pull request:
I added DataCite support using the [gbif-doi](https://github.com/gbif/gbif-doi) library (thanks to @timrobertson100 for the suggestion and for help providing me some test datacite account).

I tested minting test DOIs using `api.test.datacite.org` but in short I'll try with the production API in:
https://doi.gbif.es

I'll also send an additional PR to `ala-install` to allow the configuration of the new variables.

sbearcsiro commented on grails-app/services/au/org/ala/doi/DoiService.groovy:
```diff
@@ -24,8 +25,9 @@ import static au.org.ala.doi.util.Utils.isUuid
     GrailsApplication grailsApplication
-    AndsService andsService
-    MockService mockService
+    AndsService andsService = new AndsService()
```
Don't `new` these Service classes, leave them to Grails to inject

vjrj commented:
Off course, thanks for the review.

Sorry, I'm a java guy using a java library here, so many times I missed groovy. I added and deleted a lot of semicolons :-P

vjrj commented on grails-app/services/au/org/ala/doi/providers/DataCiteService.groovy:
```diff
@@ -0,0 +1,282 @@
+class DataCiteService extends DoiProviderService {
+    DataCiteClient dataCiteClient
+    RestJsonApiDataCiteService restDataCiteService
+    ClientConfiguration clientConf = ClientConfiguration.builder().withBaseApiUrl(
```
Done

vjrj commented on grails-app/services/au/org/ala/doi/providers/DataCiteService.groovy:
```diff
@@ -0,0 +1,282 @@
+            dcMetadata.setRightsList(rightsList)
+        }
+
+        // FIXME put lang in config
```
Done

vjrj commented:
Hi @sbearcsiro I followed all your comments. All tests, etc worked. But in some point for some reason, the pages are unstyled, without header/footer. This also happened to me testing some recents versions and snapshots from nexus via ala-install. Can you share your configuration skin and  headerAndFooter configuration?

![2020-09-05-09:06:48-screenshot](https://user-images.githubusercontent.com/180085/83847152-0bf0d480-a70c-11ea-9b43-68e467115022.png)

brucehyslop commented:
A new issue #57 has been created to address the broken style.

brucehyslop commented:
broken style has been fixed. 

@sbearcsiro are you happy with the requested changes?

nickdos commented:
Bump @sbearcsiro
